docs/DeveloperGuide.md

```diff
@@ -1,38 +1,324 @@
 # Developer Guide
 
-## Acknowledgements
+--------------------------------------------------------------------------------------------------------------------
+# Contents
+* [Acknowledgements](#acknowledgements)
+* [Getting Started For Beginners](#getting-started-for-beginners)
+* [Design](#design)
+  * [Architecture](#architecture)
+  * [UI Component](#ui-component)
+  * [Manager Component](#manager-component)
+  * [Helper Classes](#helper-classes)
+    * [Command Component](#command)
+    * [Parser Component](#parser)
+    * [Storage Component](#storage)
+    * [Validator Component](#validator)
+  * [Asset Classes](#asset-classes)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* [Implementation](#implementation)
+  * [Design Considerations](#design-considerations)
+* [Product Scope](#product-scope)
+  * [Target User](#target-user-profile)
+  * [Value Proposition](#value-proposition)
+  * [User Stories](#user-stories)
+  * [Use Cases](#use-cases)
+  * [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Appendix](#appendix)
 
-## Design & implementation
+--------------------------------------------------------------------------------------------------------------------
+## **Acknowledgements**
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+* Design and Structure of Developer Guide referenced from the
+[AB3-Developer Guide](http://se-education.org/addressbook-level3/DeveloperGuide.html).
 
+--------------------------------------------------------------------------------------------------------------------
+## **Getting Started for Beginners**
 
+Refer to the [_User Guide_](UserGuide.md).
+
+--------------------------------------------------------------------------------------------------------------------
+## **Design**
+
+### Architecture
+
+![Architecture Diagram](diagrams/OverallArch.png)
+
+The ***Architecture Diagram*** given above explains the high-level design ***HalpMi***.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+#### **Main components of the architecture**
+
+**`Duke`** has single method called `main` which is called upon launch. This initialises a new instance of a `Manager` class, and calls the `runLoop()`
+method belonging to the Manager object.
+
+[**`Assets`**](#asset-classes): Refers to a collection of classes that hold all the necessary data given by the User in
+current and past usages.
+
+The rest of the App consists of these components.
+
+* [**`Manager`**](#manager-component): The Brain.
```

Comment:
I'm not sure if this description is helpful. Maybe be a bit more explicit with what the Manager actually does? From the diagram it seems like it acts as an intermediary between the User/Duke and the rest of the components.